### ISSUE TITLE
GenericRelation \does not work in Data Migrations

### ISSUE THREAD

**Issue by Q-bart** [marazmiki/django-generic-helpers#6]
Hello!

I use this package in my project for comments model:

```
from generic_helpers.models import GenericRelationModel

class Comment(GenericRelationModel):

    user = models.ForeignKey(User)
    text = models.TextField()
    
```

And now I want to create data migration:

```
def user_read_posts(apps, schema_editor):
    User = apps.get_model("main", "User")
    Post = apps.get_model("main", "Post")
    Comment = apps.get_model("comments", "Comment")

    comments = Comment.objects.all()
    for comment in comments:
        print (comment.content_object.__class__.__name__)
        if isinstance(comment.content_object, Post):
            comment.user.read_posts.add(comment.content_object)

class Migration(migrations.Migration):

    dependencies = [
        ('main', '0039_auto_20160314_0906'),
    ]

    operations = [
        migrations.RunPython(user_read_posts),
    ]
```

But when I run it, django raises error: `AttributeError: 'Comment' object has no attribute 'content_object'`.  

When I call content_object in shell or views, it works perfectly.

What should I do??

**Comment by marazmiki** [comment]
I guess it's feature, not bug: in migrations these objects actually not real model. You can make sure yourself: add any method in your model and try to call it in migration, You also get AttributeError

**Comment by Q-bart** [comment]
Yes, i get error when try to use any method. But, when I change this (`Comment = apps.get_model("comments", "Comment")`) to `from comments.models import Comment`, `content_object` attribute works perfefectly.

So, I don't know what to think!

**Comment by marazmiki** [comment]
If you don't change both schema and data in the single migration, perhaps you can import true models and use it in migration script.

**Comment by scardine** [comment]
I have a model with a CharField that I want to change to a FK:

    # OLD: models.CharField(max_length=100)
    pax = models.ForeignKey(Traveler)

Since the content of the field is the passenger name and the pk on the Traveler model is an UUID, I get a `django.db.utils.DataError: invalid input syntax for uuid: "Silva/João"`

I tried to update the content to the correct UUID based on a generic relation from a FK:

    def forward(apps, schema_editor):
        model = apps.get_model("flight", 'Ticket')
        for ticket in model.objects.all():
            ticket.pax = ticket.flight.travels.first().traveler.pk
            ticket.save()

But this fails with `AttributeError: 'Flight' object has no attribute 'travels'`. Looks like generic relations are not visible if you get the model using `apps.get_model`.

**Comment by marazmiki** [comment]
Perhaps, the same thing here:  the ``apps`` instance you have got in a ``forward`` handler, isn't the ``django.apps.apps`` instance, it's just feigned it. The duck typing, you know.

But it's weird if ``travels`` is a native related name, not a ``Flight`` property or something like that; it migrations you can access to related managers AFAIR.

**Comment by scardine** [comment]
For now I've surrounded the problematic line in a try/except block:

    try:
        ticket.pax = ticket.flight.travels.first().traveler.pk
    except AttributeError:
        from flight.models import Flight
        ticket.pax = Flight.objects.get(pk=ticket.flight_id).travels.first().traveler_id
        
The migration runs this way, so far I can't notice any side effect.

**Comment by marazmiki** [comment]
It should work. The only thing here - make sure the order in ``travels`` is right, otherwise, you can get a wrong result. It's not an issue for the package although.

**Comment by scardine** [comment]
Thanks for answering. Regarding the order, it is always just one travel - that `.first()` could be a `.last()` as well.
